app/src/main/java/com/example/parstagram/fragments/DetailFragment.java

```diff
@@ -0,0 +1,278 @@
+        // Inflate the layout for this fragment
+        return inflater.inflate(R.layout.fragment_detail, container, false);
+    }
+
+    @Override
+    public void onViewCreated(@NonNull View view, @Nullable Bundle savedInstanceState) {
+        super.onViewCreated(view, savedInstanceState);
+        tvUsernameTop = view.findViewById(R.id.tvUsernameTop);
+        tvUsernameBottom = view.findViewById(R.id.tvUsernameBottom);
+        ivImage = view.findViewById(R.id.ivImage);
+        tvDescription = view.findViewById(R.id.tvDescription);
+        tvTime = view.findViewById(R.id.tvTime);
+        ivProfile = view.findViewById(R.id.ivProfile);
+        rvComments = view.findViewById(R.id.rvComments);
+        tvLikes = view.findViewById(R.id.tvLikes);
+        ivLike = view.findViewById(R.id.ivLike);
+        ivComment = view.findViewById(R.id.ivComment);
+        ivMessage = view.findViewById(R.id.ivMessage);
+
+        allComments = new ArrayList<>();
+
+        adapter = new CommentsAdapter(getContext(), allComments);
+        rvComments.setAdapter(adapter);
+        rvComments.setLayoutManager(new LinearLayoutManager(getContext()));
+        queryComments();
+
+        tvUsernameTop.setText(getArguments().getString("tvUsername"));
+        tvUsernameBottom.setText(getArguments().getString("tvUsername"));
+        tvDescription.setText(getArguments().getString("tvDescription"));
+        Glide.with(getContext()).load(getArguments().getString("ivImage")).into(ivImage);
+        tvTime.setText(getArguments().getString("tvTime"));
+        if (getArguments().getString("ivProfile") == null) {
+            Glide.with(getContext())
+                    .load(R.drawable.default_profile)
+                    .circleCrop()
+                    .into(ivProfile);
+        } else {
+            Glide.with(getContext())
+                    .load(getArguments().getString("ivProfile"))
+                    .circleCrop()
+                    .into(ivProfile);
+        }
+        ivLike.setImageResource(R.drawable.ufi_heart);
+        Post post = (Post) getArguments().getSerializable("post");
+        try {
+            int numQueryLikes = queryLikes(post);
+            if (numQueryLikes == 1) {
+                String text = "Liked by 1 person";
+                tvLikes.setText(text);
+            } else {
+                String text = "Liked by " + numQueryLikes + " people";
+                tvLikes.setText(text);
+            }
+        } catch (ParseException e) {
+            e.printStackTrace();
+        }
+        try {
+            if (liked(post)) {
+                ivLike.setImageResource(R.drawable.ufi_heart_active);
+                DrawableCompat.setTint(
+                        DrawableCompat.wrap(ivLike.getDrawable()),
+                        ContextCompat.getColor(getContext(), R.color.red)
+                );
+            }
+        } catch (ParseException e) {
+            e.printStackTrace();
+        }
```

Comment:
This block of code is absolutely huge. I'd recommend some refactoring or at least more spaces and comments to create better reading flow. Your code is clear tho!